### PR TITLE
[Feature] added module-unification route and route-test blueprints

### PR DESCRIPTION
**Pull request by kevinansfield**
Part of ember-cli/ember-cli#7530

This PR adds initial support for `ember g route foo` from a module unification app.

QUESTIONS:
- [x] I've had to copy the [`app/router.js` file to `src/router.js`](https://github.com/emberjs/ember.js/pull/16663/files#diff-7b356d860fa1783b996c6a9d9ce2faa6R514) in the tests. Is this approach ok for the moment? I couldn't see a way to get `emberNew` to generate a MU-style app/addon structure

### CODE REVIEW

**Comment by GavinJoyce** [comment]
> I couldn't see a way to get emberNew to generate a MU-style app/addon structure

I don't believe that there currently is a way to pass the required environment variables through `ember-cli-blueprint-test-helpers` so that it generates a MU app or addon. It would be useful to create proper MU apps though, I've added a todo item to https://github.com/ember-cli/ember-cli/issues/7530

I think this approach is fine for now though

File: blueprints/route/index.js [hunk 1]
```diff
@@ -26,48 +27,75 @@ module.exports = {
     },
   ],
 
+  filesPath: function() {
+    let filesDirectory = 'files';
+
+    if (isModuleUnificationProject(this.project)) {
+      filesDirectory = 'module-unification-files';
```
[the remark this answers is not in the document]
Comment (kevinansfield) [reply blueprints/route/index.js]: 👍 updated

**Review by GavinJoyce — APPROVED** [review]
👍 this is great, thanks @kevinansfield!

**Comment by rwjblue** [comment]
Thank you!!

**Pull request merged**